heron/proto/ckptmgr.proto

```diff
@@ -0,0 +1,200 @@
+package heron.proto.ckptmgr;
+option java_package = "com.twitter.heron.proto.ckptmgr";
+option java_outer_classname = "CheckpointManager";
+
+import "common.proto";
+import "physical_plan.proto";
+
+// Message representing information about a globally consistent
+// checkpoint
+message StatefulConsistentCheckpoint {
+  required string checkpoint_id = 1;
+  // One can add more meta data about this ckpt later
+}
+
+// message stored in the state manager by the tmaster
+message StatefulConsistentCheckpoints {
+  // An ordered list of the globally consistent checkpoints
+  // that have been snapshotted and can be recovered from.
+  // 0th element is the most recent one, 1st is the next recent
+  // and so on
+  repeated StatefulConsistentCheckpoint consistent_checkpoints = 1;
+}
+
+// tmaster -> stmgr messages
+
+// Message sent to stmgrs by the tmaster to initiate checkpointing
+message StartStatefulCheckpoint {
```

Comment:
Do the heron client/server model use a protobuf RPC interface or do they just send protobuf object bytes on raw sockets?